packages/truffle-debugger/lib/evm/reducers.js

```diff
@@ -127,7 +127,9 @@ export function callstack(state = [], action) {
       return state.concat([ {binary} ]);
 
     case actions.RETURN:
-      return state.slice(0, -1); // pop
+      //pop the stack, unless that would leave it empty (this will only happen
+      //at the end when we want to keep the last one around)
```

Comment:
This looks like a candidate for a `HACK` marker

Reply:
Heh, yeah.

Reply:
I can add that to the other instance of this pattern as well...